arangod/Pregel/Conductor.cpp

```diff
@@ -1141,3 +1122,15 @@ void Conductor::updateState(ExecutionState state) {
     _expires = std::chrono::system_clock::now() + _ttl;
   }
 }
+
+auto Conductor::changeState(conductor::StateType name) -> void {
```

Comment:
For me, `changeState` sounds as if only the state is changed. However, it is also produced and `run()` is performed. Shouldn't these three things be separated?

Reply:
The states are designed such that their run() method should be executed as soon as a state-change happened.
Like this the conductor is either running (executes the state->run() method) or waits for a message from a worker which will change the state in the state->receive method and then triggers state->run() on the new state again.
Yes, in this state-machine the states are created just when they are triggered. I chose to do that because the conductor reaches each state only once - except if a recovery happens, but I assume this will not happen too often. That this state is created is a detail which is not relevant for the caller.